webapp/app/js/perf.js

```diff
@@ -6,25 +6,273 @@
+              (userOptions.pgo && !_.contains(seriesSummary.options, 'pgo')) ||
+              (!userOptions.pgo && _.contains(seriesSummary.options, 'pgo'))) {
+              return;
+          } else {
+            seriesList.push(seriesSummary);
+
+            // add test/platform to lists if not yet present
+            if (!_.contains(platformList, seriesSummary.platform)) {
+              platformList.push(seriesSummary.platform);
+            }
+            if (!_.contains(testList, seriesSummary.name)) {
+              testList.push(seriesSummary.name);
+            }
+          } //if/else
+        }); //lists.serieslist.forEach
+
+        return {
+          seriesList: seriesList,
+          platformList: platformList,
+          testList: testList
+        };
+      }); //_getAllSeries
+    },
+
+  }
+  }]);
+
+perf.factory('isReverseTest', [ function() {
+  return function(testName) {
+    var reverseTests = ['dromaeo_dom', 'dromaeo_css', 'v8_7', 'canvasmark'];
+    var found = false;
+    reverseTests.forEach(function(rt) {
+      if (testName.indexOf(rt) >= 0) {
+        found = true;
+      }
+    });
+    return found;
+  }
+}]);
+
+
+perf.factory('getCounterMap', [ 'isReverseTest',
+  function(isReverseTest) {
+    return function(testName, originalData, newData) {
+      var cmap = {originalGeoMean: NaN, originalRuns: 0, originalStddev: NaN,
+                  newGeoMean: NaN, newRuns: 0, newStddev: NaN,
+                  delta: NaN, deltaPercentage: NaN, isEmpty: false,
+                  isRegression: false, isImprovement: false, isMinor: true};
+
+      if (originalData) {
+         cmap.originalGeoMean = originalData.geomean;
+         cmap.originalRuns = originalData.runs;
+         cmap.originalStddev = originalData.stddev;
+         cmap.originalStddevPct = ((originalData.stddev / originalData.geomean) * 100);
+      }
+      if (newData) {
+         cmap.newGeoMean = newData.geomean;
+         cmap.newRuns = newData.runs;
+         cmap.newStddev = newData.stddev;
+         cmap.newStddevPct = ((newData.stddev / newData.geomean) * 100);
+      }
+
+      if ((cmap.originalRuns == 0 && cmap.newRuns == 0) ||
+          (testName == 'tp5n summary opt')) {
+        // We don't generate numbers for tp5n, just counters
+        cmap.isEmpty = true;
+      } else {
+        cmap.delta = (cmap.newGeoMean - cmap.originalGeoMean);
+        cmap.deltaPercentage = (cmap.delta / cmap.originalGeoMean * 100);
+        if (cmap.deltaPercentage > 2.0) {
+          cmap.isMinor = false;
+          isReverseTest(testName) ? cmap.isImprovement = true : cmap.isRegression = true;
+        } else if (cmap.deltaPercentage < -2.0) {
+          cmap.isMinor = false;
+          isReverseTest(testName) ? cmap.isRegression = true : cmap.isImprovement = true;
+        }
+      }
+      return cmap;
     }
-    name = name + " " + options.join(" ");
+}]);
+
+
+perf.factory('getResultsMap', [ '$q', '$http', 'thServiceDomain', 'phSeries', 'math',
```

Comment:
Let's fold getCounterMap, getResultsMap into a factory called PhCompare -- they're only really useful in that context.